kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -206,6 +208,36 @@ object KinesisSink {
       case None => None.asRight
     }
 
+  /**
+    * Check whether an SQS queue (buffer) exists
+    * @param name Name of the queue
+    * @return Whether the queue exists
+    */
+  private def sqsQueueExists(client: AmazonSQS, name: String): Boolean =
+    try {
+      client.getQueueUrl(name)
+      true
+    } catch {
+      case _: QueueDoesNotExistException => false
+    }
+
+  private def runChecks(
+    kinesisClient: AmazonKinesis,
+    streamName: String,
+    sqs: Option[SqsClientAndName]
+  ): Either[IllegalArgumentException, Boolean] =
+    if (streamExists(kinesisClient, streamName)) {
+      true.asRight
+    } else if (sqs.isDefined && sqsQueueExists(
+                 sqs.get.sqsClient,
+                 sqs.get.sqsBufferName
+               )) {
+      true.asRight
+    } else {
+      new IllegalArgumentException(
+        s"Kinesis stream $streamName doesn't exist and no SQS buffer is configured. At least one of the two is required. If you have configured an SQS buffer, double-check if the queues specified in your config file actually exist."
+      ).asLeft
+    }
```

Comment:
```suggestion
private def runChecks(kinesisClient: AmazonKinesis, streamName: String, sqs: Option[SqsClientAndName]): Either[IllegalArgumentException, Unit] =
  sqs match {
    case Some(clientAndName) => 
      if (sqsQueueExists(clientAndName.sqsClient, clientAndName.sqsBufferName)) 
        ().asRight
      else new IllegalArgumentException(s"SQS queue ${clientAndName.sqsBufferName} is defined, but not exists").asLeft  // Assuming this is also invalid state
    case None if streamExists(kinesisClient, streamName) =>
      ().asRight
    case None =>
      new IllegalArgumentException(
        s"Kinesis stream $streamName doesn't exist and no SQS buffer is not configured. At least one of the two is required. If you have configured an SQS buffer, double-check if the queues specified in your config file actually exist."
      ).asLeft
  }
```

1. `Boolean` doesn't bear any information here
2. I think `Option.get` should not be used if there's even slight chance to avoid it
3. The orginal error message might be misleading because in final `else` we don't know if SQS buffer is configured, but not exists or not configured

Reply:
> I think Option.get should not be used if there's even slight chance to avoid it

You purist.

I've taken this onboard.